OpenSpace/OpenSpace-WebGui
# Scene Tree

Implements the scene tree part of the scene menu, including a new design that splits the scene graph node view from the tree. 

**OBS! Note that no work has been done on the components for the Properties in this PR, to limit the scope of the changes. I will create another PR with the updated properties** 

To test: 

1. The behavior of the tree, related features, and the new design. 
2. Keyboard navigation!  This needs some improvement, but I would like to discuss how to best implement it
3. Code structure and code itself. Does ti make sense how files are located and split up, etc.. 
4. New features:  Filter by tags, and new information panel for scene graph nodes (including tags and copy-buttons)

Some TODOs (that need discussion and/or could be fixed later):
- [ ] Decide how SGNs should look in other lists (such as the Exoplanet panel and ScreenSpace panel). Might for example need a "Show me this node in the Scene Tree" feature, at least for exoplanets. 
- [x] Fade checkbox implementation is temporary. We should decide on an alternative design that feels more "snappy". 
- [ ] When opening a new window, it should automatically be given focus (to make it easier to navigate using a keyboard). It should also be easy to close (e.g. using ESC ?), and when doing so, return focus to the previous node. 
- [x] Keyboard navigation on the tree - how should it work? E.g. when on a Scene Graph Node, do we want to toggle it or open the window with settings when for example pressing the ENTER key?
- [ ] Virtualize the tree, based on component from https://github.com/OpenSpace/OpenSpace-WebGui/pull/22. It is really slow when all nodes are expanded... (partly because the tree is rendered as soon as an item is expanded). I have started looking at other alternatives a bit, an this seems like a promising option: https://github.com/brimdata/react-arborist
- [ ] Add a way to reorder the layers that's accessible (e.g. using keyboard input)

There are also a lot of style-related things to decide, such as aligning things nicely in the tree and deciding on sizes that make it easier to see the content of the tree. For now, I have tried to keep it as simple as possible so that we can update those things later on.
